From 63bb02e2299822a13804b11e4cf218ea4213af6a Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Wed, 29 Oct 2014 15:54:58 -0400
Subject: [ethernet] mlx4: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate

Message-id: <1988631cebca6d14dab08201fbab4a25fe2ae59a.1414597479.git.avadai@redhat.com>
Patchwork-id: 98514
O-Subject: [Patch RHEL7.1 7/9] net: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate
Bugzilla: 1060221
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Jiri Pirko <jiri@resnulli.us>

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 537fae0101c8853abb52136788173dde74b9d1e9)
Signed-off-by: Amir Vadai <avadai@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

Conflicts:
 drivers/net/ethernet/intel/igb/igb_ethtool.c

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c b/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
index 142f0c3..6e7851b 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
@@ -391,8 +391,8 @@ static int mlx4_en_get_settings(struct net_device *dev, struct ethtool_cmd *cmd)
   ethtool_cmd_speed_set(cmd, priv->port_state.link_speed);
   cmd->duplex = DUPLEX_FULL;
  } else {
-  ethtool_cmd_speed_set(cmd, -1);
-  cmd->duplex = -1;
+  ethtool_cmd_speed_set(cmd, SPEED_UNKNOWN);
+  cmd->duplex = DUPLEX_UNKNOWN;
  }
 
  if (trans_type > 0 && trans_type <= 0xC) {
-- 
1.7.1